packages/synapse-react-client/src/components/ChallengeDataDownload/ChallengeDataTable.tsx

```diff
@@ -0,0 +1,486 @@
+                let { currentSelectedVersion } = rowData
+                if (!isDisabled) {
+                  if (
+                    isVersionableEntity &&
+                    versionSelection === VersionSelectionType.REQUIRED &&
+                    currentSelectedVersion == null &&
+                    Object.prototype.hasOwnProperty.call(
+                      rowData,
+                      'versionNumber',
+                    )
+                  ) {
+                    currentSelectedVersion = (rowData as EntityHeader)
+                      .versionNumber
+                    // Note that here we aren't handling the case where the header doesn't have a version, e.g. a search result
+                    // That case is actually handled by the VersionRenderer, which has an effect that will toggle the selection after data is fetched.
+                  }
+
+                  toggleSelection({
+                    targetId: id,
+                    targetVersionNumber:
+                      currentSelectedVersion === NO_VERSION_NUMBER
+                        ? undefined
+                        : currentSelectedVersion,
+                  })
+                }
+              },
+            }}
+            onEndReached={() => {
+              if (hasNextPage && fetchNextPage && !isFetchingNextPage) {
+                fetchNextPage()
+              }
+            }}
+            emptyRenderer={
+              isLoading
+                ? LoadingRenderer
+                : () => (
+                    <EmptyRenderer
+                      noResultsPlaceholder={noResultsPlaceholder}
+                    />
+                  )
+            }
+          >
+            {showSelectColumn && (
+              <Column<EntityFinderTableViewRowData>
+                key="isSelected"
+                title=""
+                minWidth={50}
+                maxWidth={50}
+                width={50}
+                dataKey="isSelected"
+                headerRenderer={SelectAllCheckboxRenderer}
+                cellRenderer={DetailsViewCheckboxRenderer}
+              />
+            )}
+            <Column<EntityFinderTableViewRowData>
+              key="type"
+              title=""
+              minWidth={45}
+              maxWidth={45}
+              width={45}
+              dataKey="entityType"
+              align="center"
+              cellRenderer={TypeIconRenderer}
+            />
+            <Column<EntityFinderTableViewRowData>
+              key={SortBy.NAME}
+              title="File Name"
+              width={500}
+              sortable={sort != null}
+              resizable={true}
+              cellRenderer={NameRenderer}
+            />
+            <Column<EntityFinderTableViewRowData>
+              key={'SIZE'}
+              title="Size"
+              width={200}
+              sortable={false}
+              resizable={true}
+              cellRenderer={SizeRenderer}
+            />
```

Comment:
Much of this file is lifted direction from `DetailsView`. And while it pained me to do this, this was the best approach given our time constraints. I believe there's an opportunity here for future re-consolidation.